src/datasets.js

```diff
@@ -260,59 +261,52 @@ export default {
         const {
             baseUrl,
             datasetId,
-            offset,
-            limit,
-            fields,
-            omit,
-            unwind,
-            desc,
-            bom,
-            attachment,
-            delimiter,
             disableBodyParser,
-            xmlRoot,
-            xmlRow,
-            skipHeaderRow,
-            token,
         } = options;
 
         checkParamOrThrow(baseUrl, 'baseUrl', 'String');
         checkParamOrThrow(datasetId, 'datasetId', 'String');
-        checkParamOrThrow(limit, 'limit', 'Maybe Number');
-        checkParamOrThrow(offset, 'offset', 'Maybe Number');
-        checkParamOrThrow(fields, 'fields', 'Maybe Array');
-        checkParamOrThrow(omit, 'omit', 'Maybe Array');
-        checkParamOrThrow(unwind, 'unwind', 'Maybe String');
-        checkParamOrThrow(desc, 'desc', 'Maybe Boolean');
-        checkParamOrThrow(bom, 'bom', 'Maybe Boolean');
-        checkParamOrThrow(attachment, 'attachment', 'Maybe Boolean');
-        checkParamOrThrow(delimiter, 'delimiter', 'Maybe String');
-        checkParamOrThrow(xmlRoot, 'xmlRoot', 'Maybe String');
-        checkParamOrThrow(xmlRow, 'xmlRow', 'Maybe String');
-        checkParamOrThrow(skipHeaderRow, 'skipHeaderRow', 'Maybe Number');
-        checkParamOrThrow(token, 'token', 'Maybe String');
+        checkParamOrThrow(disableBodyParser, 'disableBodyParser', 'Maybe Boolean');
+
+        // Query params:
+        checkParamOrThrow(options.token, 'token', 'Maybe String');
+        checkParamOrThrow(options.offset, 'offset', 'Maybe Number');
+        checkParamOrThrow(options.limit, 'limit', 'Maybe Number');
+        checkParamOrThrow(options.fields, 'fields', 'Maybe [String]');
+        checkParamOrThrow(options.omit, 'omit', 'Maybe Array');
+        checkParamOrThrow(options.delimiter, 'delimiter', 'Maybe String');
+        checkParamOrThrow(options.unwind, 'unwind', 'Maybe String');
+        checkParamOrThrow(options.xmlRoot, 'xmlRoot', 'Maybe String');
+        checkParamOrThrow(options.xmlRow, 'xmlRow', 'Maybe String');
+        checkParamOrThrow(options.format, 'format', 'Maybe String');
+
+        // Booleans query params:
+        checkParamOrThrow(options.desc, 'desc', 'Maybe Boolean');
+        checkParamOrThrow(options.bom, 'bom', 'Maybe Boolean');
+        checkParamOrThrow(options.attachment, 'attachment', 'Maybe Boolean');
+        checkParamOrThrow(options.skipHeaderRow, 'skipHeaderRow', 'Maybe Boolean');
+
+        // Pick query params.
+        const query = _.pick(options, 'offset', 'limit', 'fields', 'omit', 'delimiter', 'unwind', 'xmlRoot', 'xmlRow', 'format', 'token');
+
+        // Add Boolean query params.
+        if (options.skipHeaderRow) query.skipHeaderRow = 1;
+        if (options.desc) query.desc = 1;
+        if (options.bom) query.bom = 1;
+        if (options.attachment) query.attachment = 1;
+
+        if (query && query.fields) query.fields = query.fields.join(',');
```

Comment:
`query` will always be at least an empty object.

Reply:
Done